### ISSUE TITLE
Add versioning of packages to CI / deployment pipeline

### ISSUE THREAD

**Issue by oriondean** [ScottLogic/StockFlux#1118]
Backend pipeline:
- [x] Append build number to version in root package.json
- [x] Copy version number to package.json files of other deployed modules before packing up
- [x] Upload packaged artifacts to versioned folder in private S3 bucket
- [x] Deploy via CloudFormation, passing through version, stage and other required parameters
- [x] Organise restrictive permissions for AWS CI user to allow it to do only what it needs to do
- [x] Integrate the distribution alias with CloudFront in CloudFormation template, ensuring the domain is passed through as a header to API Gateway and Lambdas
- [x] Investigate using CloudFormation `If` intrinsic function with [`AWS::NoValue`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html#cfn-pseudo-param-novalue) to conditionally omit properties in certain conditions i.e. see if we can omit the distribution alias and stop the header from being passed through if we're not deploying the master/production stage
- [x] Ensure that stack updates actually trigger Lambda code changes
- [x] Allow for each environment to have a different distribution alias, automatically populated by CI based on stage/branch name
- [x] Create tag in GitHub for version once deployment is complete

When a stage does not have a custom domain (all stages except master/production) then the CloudFront domain should be passed through the headers. This is to be implemented as part of a separate issue #
